### PR TITLE
feat: use latest parser and support spec 2.1 validation

### PR DESCRIPTION
**Pull request by derberg**
so far it is just parser 1.6 because 1.7 is not yet released, changes in dummy.yml are for now commented out

### CODE REVIEW

**Review by magicmatatjahu — DISMISSED** [review]
LGTM

File: test/docs/dummy.yml [hunk 1]
```diff
@@ -188,12 +188,15 @@ components:
       payload:
         $ref: "#/components/schemas/dummyInfo"
       examples:
-        - headers:
+        - #name: option1example
```
Comment (magicmatatjahu) [inline test/docs/dummy.yml:191]: It should be uncommented also another comments.

Comment (derberg) [reply test/docs/dummy.yml:191]: omg!

**Review by magicmatatjahu — APPROVED** [review]

**Review by magicmatatjahu — APPROVED** [review]
LGTM 🚀

**Pull request merged**